iFixit/react-commerce
# Tests: Collections page add tests

## Summary

Adds 2 new tests to the collection (`/Tools`) page.
- The first test clicks on the next page pagination button and checks if the window scrolls up. The current behavior when clicking the next page button, the page scrolls up to the search bar. This test checks if the search bar is visible in the viewport.
   - Added cypress [command](https://stackoverflow.com/a/58713927) to check if an element is in viewport.
- The second test checks the product view by clicking on grid/list view button.

Added a few `data-testid` attributes on some pages.

## QA 

Please make sure `E2E tests / cypress-run` ci check passes.

Connects: https://github.com/iFixit/ifixit/issues/43590
